src/components/toolbar/toolbar.js

```diff
@@ -102,20 +118,41 @@ const Toolbar = ({
    * Set selected options for chip display.
    *
    * @param {object} params
-   * @param {string|Array<string>} params.value
-   * @returns {Array}
+   * @param {string|Array<string|{name:string, isDisplayValueOnly:boolean}>} params.value
+   * @returns {Array<string>}
    */
   const setSelectedOptions = ({ value: filterName } = {}) => {
-    const filterValues =
-      (Array.isArray(filterName) && filterName.map(filter => ({ name: filter, value: toolbarFieldQueries[filter] }))) ||
-      (typeof toolbarFieldQueries[filterName] === 'string' && [
-        { name: filterName, value: toolbarFieldQueries[filterName] }
-      ]) ||
-      [];
+    const filterValues = [];
+
+    if (typeof toolbarFieldQueries[filterName] === 'string') {
+      filterValues.push({ name: filterName, value: toolbarFieldQueries[filterName] });
+    }
+
+    if (Array.isArray(filterName)) {
+      filterValues.push(
+        ...filterName.map(filter => {
+          if (typeof filter === 'string') {
+            return { name: filter, value: toolbarFieldQueries[filter] };
+          }
+
+          const { isDisplayValueOnly, name } = filter;
+          return {
+            isDisplayValueOnly,
+            name,
+            value: toolbarFieldQueries[name]
+          };
+        })
+      );
+    }
```

Comment:
future: refactor should include making some flavor of this "setting selected options" dep injected for testing